### ISSUE TITLE
Update PlayerMovement.cs and the Player GameObject Composition Part 1

### ISSUE THREAD

**Issue by duckbossa** [jamesantiporda/Encore#1]
# Foreword

### TLDR: There are portions of `PlayerMovement.cs`, that can be done with Unity's built in physics without having the need to write custom code.

While there are a lot of things that can be improved in the PlayerMovement.cs script, tackling them in one go is neither time efficient, and may dilute the understanding of _why_ they need improving in the first place. In the this and the following issues to be posted, I will be picking out certain portions of the project that can be refactored in sizes that should be small enough to do within a branch, and hopefully point out why the existing code needs changing. 

# Player Updates

## Invisible bounding box

Take a look at the following snippets in PlayerMovement.cs

https://github.com/jamesantiporda/Encore/blob/32228971a9ef4245c4845431f7978ee7bff238c5/Assets/Scripts/New%20Scripts/PlayerMovement.cs#L210-L225

The above code is used to ensure that the player is within some bound in a level. From an outsider's point of view, it is unclear _why_ 
the number the the player position needs to be within the constants defined the way the are without having to look into how the level scene is composed. Upon inspecting any arbitrary level scene in the project, the above coordinates represent a bounding box in which the player cannot move beyond.

While this solution currently works for the existing levels, future levels may allow the player to move deeper into the level. Allowing for this particular implementation makes your project inflexible for future updates.

## Solution

The solution for this is quite simple really. We just need to **remove the code altogether**.  How does removing the code solve the problem ? Well, the solution lies in using Unity's already built in collision system to do the bounding checks for us. 

![image](https://github.com/user-attachments/assets/6777b1b9-ab38-4025-8960-cd84b340cbfc)

- In your player GameObject, remove the **IsTrigger** property on the box 2d collider.
- In the rigidbody2d of your player, add the freeze rotation constraint. This will ensure your character stays in the same direction even if they get hit.
![image](https://github.com/user-attachments/assets/c993c1a2-a951-46b8-aa00-5612984a700b)
- In any of your levels, add an empty GameObject and name it MapBounds, make sure it's position is reset to 0,0,0 and add 4 new empty ChildObjects inside of it. 
- In each of the child object, add a rect2d collider that is not set into trigger, and place them how you would want the bounds of your game to be defined. Below is a sample of how i did it. 

![image](https://github.com/user-attachments/assets/9db900cf-56b7-4e0c-b693-5b449f8ea633)

- By doing this simple update to the way you structure your player GameObject and your map, you have now just gotten rid of the need write custom code for your game that can be solved by using the built-in systems of Unity.

## Notes
After doing the updates myself, the game becomes unplayable because of how the projectiles are implemented in the game. There are multiple ways of dealing with this, but the way I would recommend solving the issue is to learn about [Unity's Collision Matrix](https://docs.unity3d.com/Manual/LayerBasedCollision.html).

-  Create custom collision layers for your Map, Player, Enemy, Projectiles
- Update the relevant game objects to use the layers you've created
- Update the CollisionMatrix in the Edit -> ProjectSettings -> Physics2D -> Layer Collision Matrix
